### PR TITLE
Excluded rules summary

### PR DESCRIPTION
**Pull request by dogversioning**
This adds a series of tables designed to help debug study configs by providing a list of rules that were excluded, but were considered potential candidates for exclusion.

This adds a new study arg, debug, which is used to enable these tables. They are off by default.

### CODE REVIEW

File: cumulus_library_opioid/manifest.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ file_names = [
     "vocab/static_builder.py",
     "vocab/rxnorm_vsac_builder.py",
     "vocab/additional_rules_builder.py",
+    "vocab/excluded_rules_summary.py",
```
Comment (mikix) [inline cumulus_library_opioid/manifest.toml:8]: that's the wrong filename, right?  I think you want `excluded_rules_builder.py`

File: cumulus_library_opioid/vocab/template_sql/create_excluded_rela_types.sql.jinja [hunk 2]
```diff
@@ -0,0 +1,12 @@
+CREATE VIEW {{ prefix }}{{ steward }}_excluded_{{ name }} AS 
+SELECT 
+    rxcui1,
+    tty1,
+    rel,
+    rela,
+    rxcui2,
+    tty2,
+    str2
```
Comment (mikix) [inline cumulus_library_opioid/vocab/template_sql/create_excluded_rela_types.sql.jinja:9]: Just as an aside, these field names are all meaningless to someone not steeped in this repo. I know some of this is inherited. But phew.

File: tests/test_excluded_rules.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        {'name' : 'opioid__acep_excluded_is_a','rows':2},
+        {'name' : 'opioid__acep_excluded_ingredient','rows':6},
+        {'name' : 'opioid__acep_excluded_dose_form','rows':4},
+        {'name' : 'opioid__acep_excluded_refresh','rows':2},
+        {'name' : 'opioid__acep_excluded_form','rows':2},
+        {'name' : 'opioid__acep_excluded_other','rows':1},
+    ]:
+        res = cursor.execute (f"Select rela from {table_conf['name']}").fetchall()
+        assert len(res) == table_conf['rows']
+        assert len(res) == len(set(res))
```
Comment (mikix) [inline tests/test_excluded_rules.py:45]: I'm always slightly nervous about _only_ checking lengths of things. Can we have a test that maybe does a single spot check of the actual results too?

**Review by mikix — APPROVED** [review]

File: cumulus_library_opioid/manifest.toml (hunk 1 above)
Comment (dogversioning) [reply cumulus_library_opioid/manifest.toml:8]: changed it everywhere but here, thanks

File: cumulus_library_opioid/vocab/template_sql/create_excluded_rela_types.sql.jinja (hunk 2 above)
Comment (dogversioning) [reply cumulus_library_opioid/vocab/template_sql/create_excluded_rela_types.sql.jinja:9]: i :hate: these names. but it's either exactly an NIH label, or an NIH label with a number, so it makes it easier to look up in their documentation. 

it's certainly an artifact of a bygone age.

File: tests/test_excluded_rules.py (hunk 3 above)
Comment (dogversioning) [reply tests/test_excluded_rules.py:45]: i thought you would have this comment - i was less concerned about it due to the tightly controlled synthetic test data, but i can add some additional validation.

**Pull request merged**